core/Symbols.cc

```diff
@@ -1112,6 +1112,116 @@ TypePtr Symbol::sealedSubclassesToUnion(const GlobalState &gs) const {
     return result;
 }
 
+// Record a required ancestor for this class of module
+void Symbol::recordRequiredAncestor(MutableContext ctx, SymbolRef ancestor, Loc loc) {
+    ENFORCE(this->isClassOrModule(), "Symbol is not a class or module: {}", this->show(ctx));
+
+    // We store the required ancestors into a fake property called `<required-ancestors>`
+    auto ancestors = this->findMember(ctx, Names::requiredAncestors());
+    if (!ancestors.exists()) {
+        ancestors = ctx.state.enterMethodSymbol(loc, this->ref(ctx), Names::requiredAncestors());
+        ancestors.data(ctx)->locs_.clear(); // Remove the original location
```

Comment:
`ctx.state.enterMethodSymbol(loc, this->ref(ctx), Names::requiredAncestors())` stores the `loc` in the Symbol's `locs` array. Let's remove it to avoid some offsetting later.